Test/PreFlightTest.cpp

```diff
@@ -508,6 +508,26 @@ void TestNewInstanceAvailable()
   }
 }
 
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+void TestGetNameOfClass()
+{
+  qDebug() << "-------------- TestGetNameOfClass ------------------------------";
+  FilterManager* fm = FilterManager::Instance();
+  FilterManager::Collection factories = fm->getFactories();
+  QMapIterator<QString, IFilterFactory::Pointer> iter(factories);
+  while(iter.hasNext())
+  {
+    iter.next();
+    IFilterFactory::Pointer factory = iter.value();
+    AbstractFilter::Pointer filter = factory->create();
+
+    DREAM3D_REQUIRED(iter.key(), ==, filter->getNameOfClass())
+  }
+}
+
```

Comment:
If we want to require getNameOfClass() to be implemented, we could make it pure virtual in AbstractFilter.